### ISSUE TITLE
Title of vid in fullscreen v2

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [FreeTubeApp/FreeTube#857]
**This enhancement is related to #694** 

Hi @TDDR i saw the implementation while watching this vid:
https://youtu.be/ipJb4mB5bMA

it seemed fine in the beginning of the vid, but when a white background went by in the vid i couldn't see the title anymore when i moved with my mouse. The title should be clear to see at all times.

![Screenshot (9)](https://user-images.githubusercontent.com/73130443/101463013-c91aee00-3934-11eb-8bd2-2155dd186620.png)

I also checked what the max character length is of a YT vid and it is 100(including spaces), i searched a vid with max characters and this is what i got.

![Screenshot (10)](https://user-images.githubusercontent.com/73130443/101481162-9e3c9400-394c-11eb-86a2-955e7d85536a.png)

Now that we know that this is the max length of characters, i think it is better to resize the font to fit all the characters so this wont happen again. 

This is the link to that vid:
https://youtu.be/YkL3P0p_wMk

There are also people that abuse Unicode characters in video titles like this one 𒐫 (this is one character) and paste it 100times. YT just cuts the spam of characters off in full screen and replaces it with ... if it gets too long. We also need that in place.

![Screenshot (11)](https://user-images.githubusercontent.com/73130443/101638757-643fc080-3a26-11eb-8990-b2d24c935f4a.png)

Edit: After some investigation i found out that the widest character on a standard keyboard is the '@'.

Now u can paste 100 characters next to each other to see how much u need to resize the font. I hope the font doesnt get too small.

Optional: Find out what characters are the widest in foreign languages such as Mandarin, Arabic, Hindi and compare them to the widest character of a standard keyboard.
If the character of a foreign language is wider then the widest character of a standard keyboard.
Resize the font.

2nd Optional: Do the first option but then with emoji's.

P.S.
I know these things sound a little extreme, but its better to kind of stress test FreeTube in these situations, because there are some really retard people on YT. They dont break YT with it, SO WE WONT LET THEM BREAK THE ALMIGHTY FREETUBE!

**Comment by GilgusMaximus** [comment]
Thanks for creating a new issue.
So for the visibility of the video title, a simple solution like adding a black border around the text would be an easy fix.
Because recognizing where the video is light and dark for all characters could slow down the app significantly for every frame of the video. 
Another possibility would be to add a medium transparent darker overlay on the top of the video, similar to YouTube. This allows the white text to be visible at any time as well.

Regarding the font size, this should be taken into account

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
The Youtube kind of solution sounds better then putting a black border around it.
Because if the border is in place, i can imagine a few scenario's where u won't see the title (but this is just speculation).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Is it possible to implement the same solution that is used for the mediabar in fullscreen?
Everything in there is white just like the title and that doesn't disappear.

**Comment by GilgusMaximus** [comment]
From what I think it was implemented, we should be able to change the background color of the text field. But I'd have to look more into this. Unfortunately I'm currently quite busy.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@GilgusMaximus i noticed i put multiple problems into this one issue. Do u think its better to split them up in separate issues.
It would look something like this:

1. Title is not visible in fullscreen when a white background is being displayed.
2. Display all characters of the title in one line.

**Comment by GilgusMaximus** [comment]
Good idea. I'd change this issue to 'Title not visible on white background videos' and open a new one for the long video titles.

Thanks for checking the old issues for multiple requests, appreciate it :)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
No problem i try to help out where i can :)
I Created #1348.

I have some trouble with the creation of the second issue tho. I seems like back in the day i put allot of unrelated stuff in the issue regarding the long video title. 

Could u help me out to see what is related and what is not in this issue. I edited the issue, removed all the info of #1348. So that u get a better overview of the remaining things.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @GilgusMaximus just pinging u for visibility. I get that my last comment on this might have been buried on allot of other of ur notifications. Would really appreciate ur help 😄.
